metron-streaming/Metron-Solr/src/main/java/org/apache/metron/writer/solr/MetronSolrClient.java

```
@@ -0,0 +1,83 @@
+ * limitations under the License.
+ */
+package org.apache.metron.writer.solr;
+
+import org.apache.log4j.Logger;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.CloudSolrClient;
+import org.apache.solr.client.solrj.request.QueryRequest;
+import org.apache.solr.common.params.CollectionParams;
+import org.apache.solr.common.params.ModifiableSolrParams;
+import org.apache.solr.common.util.NamedList;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MetronSolrClient extends CloudSolrClient {
+
+  private static final Logger LOG = Logger.getLogger(MetronSolrClient.class);
+
+
+  public MetronSolrClient(String zkHost) {
+    super(zkHost);
+  }
+
+  public void createCollection(String name, int numShards, int replicationFactor) {
+    if(!listCollections().contains(name)) {
+      try {
+        request(getCreasteCollectionsRequest(name, numShards, replicationFactor));
+      } catch (SolrServerException | IOException e) {
+        LOG.error(e, e);
+      }
+    }
+  }
+
+  public QueryRequest getCreasteCollectionsRequest(String name, int numShards, int replicationFactor) {
+    ModifiableSolrParams params = new ModifiableSolrParams();
+    params.set("action", CollectionParams.CollectionAction.CREATE.name());
+    params.set("name", name);
+    params.set("numShards", numShards);
+    params.set("replicationFactor", replicationFactor);
+    params.set("collection.configName", name);
+    QueryRequest request = new QueryRequest(params);
+    request.setPath("/admin/collections");
+    return request;
+  }
+
+  public List<String> listCollections() {
+    List<String> collections = new ArrayList<>();
+    try {
+      NamedList<Object> response = request(getListCollectionsRequest(), null);
+      collections = (List<String>) response.get("collections");
+    } catch (SolrServerException | IOException e) {
+      LOG.error(e, e);
```

Can we bubble this exception up?